src/main/flight/interpolated_setpoint.c

```diff
@@ -0,0 +1,143 @@
+    const float ffMaxRateScale = pidProfile->ff_max_rate_limit * 0.01f;
+    ffLookaheadLimit = pidProfile->ff_lookahead_limit * 0.0001f;
+    ffSpread = pidProfile->ff_spread;
+    for (int i = 0; i < XYZ_AXIS_COUNT; i++) {
+        ffMaxRate[i] = applyCurve(i, 1.0f);
+        ffMaxRateLimit[i] = ffMaxRate[i] * ffMaxRateScale;
+    }
+}
+
+FAST_CODE_NOINLINE float interpolatedSpApply(int axis, float pidFrequency, bool newRcFrame) {
+    const float rawSetpoint = getRawSetpoint(axis);
+    const float rawDeflection = getRawDeflection(axis);
+
+    float pidSetpointDelta = 0.0f;
+    static int iterationsSinceLastUpdate[XYZ_AXIS_COUNT];
+    if (newRcFrame) {
+
+        setpointReservoir[axis] -= iterationsSinceLastUpdate[axis] * setpointChangePerIteration[axis];
+        deflectionReservoir[axis] -= iterationsSinceLastUpdate[axis] * deflectionChangePerIteration[axis];
+        iterationsSinceLastUpdate[axis] = 0;
+
+        // get the number of interpolation steps either dynamically based on RX refresh rate
+        // or manually based on ffSpread configuration property
+        if (ffSpread) {
+            interpolationSteps[axis] = (uint16_t) ((ffSpread + 1.0f) * 0.001f * pidFrequency);
+        } else {
+            interpolationSteps[axis] = (uint16_t) ((currentRxRefreshRate + 1000) * pidFrequency * 1e-6f + 0.5f);
+        }
+
+        // interpolate stick deflection
+        deflectionReservoir[axis] += rawDeflection - prevRawDeflection[axis];
+        deflectionChangePerIteration[axis] = deflectionReservoir[axis] / interpolationSteps[axis];
+        const float projectedStickPos =
+            rawDeflection + deflectionChangePerIteration[axis] * pidFrequency * ffLookaheadLimit;
+        projectedSetpoint[axis] = applyCurve(axis, projectedStickPos);
+        prevRawDeflection[axis] = rawDeflection;
+
+        // apply linear interpolation on setpoint
+        setpointReservoir[axis] += rawSetpoint - prevRawSetpoint[axis];
+        extern float ffBoostFactor;
+        if (ffBoostFactor != 0.0f) {
+            const float speed = rawSetpoint - prevRawSetpoint[axis];
+            if (fabsf(rawSetpoint) < 0.95f * ffMaxRate[axis] || fabsf(3.0f * speed) > fabsf(prevSetpointSpeed[axis])) {
+                const float setpointAcc = speed - prevSetpointSpeed[axis];
+                setpointReservoir[axis] += ffBoostFactor * setpointAcc;
+            }
+            prevSetpointSpeed[axis] = speed;
+        }
+
+        setpointChangePerIteration[axis] = setpointReservoir[axis] / interpolationSteps[axis];
+
+        prevRawSetpoint[axis] = rawSetpoint;
+        
+        if (axis == FD_ROLL) {
+            DEBUG_SET(DEBUG_FF_INTERPOLATED, 0, rawDeflection * 100);
+            DEBUG_SET(DEBUG_FF_INTERPOLATED, 1, projectedStickPos * 100);
+            DEBUG_SET(DEBUG_FF_INTERPOLATED, 2, projectedSetpoint[axis]);
+        }
+    }
+
+    if (interpolationSteps[axis]) {
+        iterationsSinceLastUpdate[axis]++;
```

Comment:
You can store expected number of iteration and calculate this when necessary

Reply:
That isn't really an option since due to jitter there may not always be the same number of steps.

Reply:
The idea was to store last expected number of iterations and update just one counter, not two

Reply:
Oh - you just mean 
```
    if (iterationsSinceLastUpdate[axis]++ < interpolationSteps[axis]) {
        pidSetpointDelta = setpointChangePerIteration[axis];
    }
```
Yeah that would eliminate the count down step, true. Probably a bit more efficient. If you confirm I'll make the change.

Reply:
Yes ... but see  comment above ..

Reply:
Would you still like this change to be made?

Reply:
It's mostly minor optimization, your proposal seems a bit easier to read, so yes ..

Reply:
(In my first comment I meant to decrement interpolationSteps[] and calculate totalInterpolationSteps[] - interpolationSteps[] when iterationsSinceLastUpdate is needed. But your way is better)